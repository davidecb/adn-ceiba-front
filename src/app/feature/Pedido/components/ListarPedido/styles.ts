import styled from 'styled-components';

export const PedidoContainer = styled.div`
  width: 80%;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  @media (max-width: 900px) {
    width: 100%;
  }
`;

export const FormularioContainer = styled.div`
  position: fixed;
  top: 0;
  z-index: 100;
  background: rgba(0, 0, 0, 0.95);
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
