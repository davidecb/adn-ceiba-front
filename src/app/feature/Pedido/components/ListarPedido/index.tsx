import * as PropTypes from 'prop-types';
import * as React from 'react';
import { FormularioContainer, PedidoContainer } from './styles';
import { Pedido } from '../../models/Pedido';
import { TarjetaProductoSolicitado } from '../TarjetaProductoSolicitado';

export interface ListaPedidoProps {
  pedido: Pedido;
  onClickFinalizarPedido: (pedido: Pedido) => void;
  eliminarProductoSolicitado: (idProductoSolicitado: number) => void;
}

export const ListaPedido: React.FC<ListaPedidoProps> = ({
  pedido,
  onClickFinalizarPedido,
  eliminarProductoSolicitado,
}) => {

  const [finalizarPedidoClicked, setFinalizarPedidoClicked] = React.useState(false);
  return (
    <PedidoContainer>
      {pedido.productosSolicitados && pedido.productosSolicitados.map((producto, index) => {
        return (
          <TarjetaProductoSolicitado 
              key={index} 
              productoSolicitado={producto}
              onClickPersonalizarProducto={()=>{
                
              }}
              eliminarProductoSolicitado={(idProductoSolicitado: number) => {
                eliminarProductoSolicitado(idProductoSolicitado);
              }}
          />
        );
      })}      
    </PedidoContainer>
  );
};

ListaPedido.propTypes = {
  pedido: PropTypes.any.isRequired,
  onClickFinalizarPedido: PropTypes.func.isRequired,
  eliminarProductoSolicitado: PropTypes.func.isRequired,
};
