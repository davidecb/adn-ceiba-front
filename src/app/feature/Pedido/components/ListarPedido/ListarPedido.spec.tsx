import * as React from 'react';
import { RenderResult, render } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { ListarPedido } from './../ListarPedido';

describe('ListarPedido Test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof ListarPedido> & {
    onClickFinalizarPedido: SinonStub;
    eliminarProductoSolicitado: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      pedido: {
        id: 1,
        numeroPedido: '123456789',
        direccion: '',
        cliente: '',
        estado: '',
        costo: 10000,
        tiempo: 30,
        productosSolicitados:[{
          id: 1,
          pedido: 1,
          cantidad: 2,
          productoSolicitado: {
            material: 'PLA',
            color: 'negro',
            acabado: {
              pulido: false,
              pintado: false,
              barnizado: false
            },
            urgencia: false,
            producto: {
              id: 1,
              nombre: 'producto testing',
              costo: 10000,
              tiempo: 30,
              imagen: 'imagen.jpg'
            }
          }
        }]
      },
      onClickFinalizarPedido: stub(),
      eliminarProductoSolicitado: stub(),
    };
    componentWrapper = render(<ListarPedido {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });
});

export {};
