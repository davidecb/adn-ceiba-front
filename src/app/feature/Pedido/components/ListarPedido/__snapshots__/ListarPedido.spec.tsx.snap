// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListarPedido Test should match snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm fLEotY"
  >
    <div
      class="sc-AxiKw dsFtPB"
    >
      <div
        class="sc-AxhCb jeWGrZ"
      >
        ‚ùå
      </div>
      <img
        class="sc-fzozJi hbgqHR"
        src="imagen.jpg"
      />
      <div
        class="sc-AxhUy bkJSOX"
      >
        producto testing
      </div>
      <span
        class="sc-AxgMl cwFzxf"
      >
        <b>
          costo/unidad:
        </b>
         $
      </span>
      <span
        class="sc-AxheI hqpeRh"
      >
        <b>
          tiempo/unidad:
        </b>
         
         min.
      </span>
      <div
        class="sc-AxmLO bpTpsY"
      >
        <div>
          <b>
            material:
          </b>
           
          PLA
        </div>
        <div>
          <b>
            color:
          </b>
           
          negro
        </div>
        <div>
          <b>
            acabado:
          </b>
          <div
            class="sc-Axmtr qsoLG"
          >
            <div>
              pulido: 
              <input
                type="checkbox"
              />
            </div>
            <div>
              pintado: 
              <input
                type="checkbox"
              />
            </div>
            <div>
              barnizado: 
              <input
                type="checkbox"
              />
            </div>
          </div>
        </div>
        <div>
          <b>
            urgencia:
          </b>
          <input
            type="checkbox"
          />
        </div>
        <div>
          <b>
            cantidad:
          </b>
           
          2
        </div>
      </div>
      <span
        class="sc-AxgMl cwFzxf"
      >
        <b>
          costo total:
        </b>
         $
        NaN
      </span>
      <span
        class="sc-AxheI hqpeRh"
      >
        <b>
          tiempo total:
        </b>
         
        NaN
         min.
      </span>
    </div>
  </div>
</div>
`;
